### PR TITLE
Feature/618/22 dpf http system tests

### PR DESCRIPTION
**Pull request by cpeeyush**
- System tests for DPF Http extension.
- Transfer result endpoint in DataPlaneTransferController.
- Transfer results are stored in-memory DataPlaneStore. 
- Usage mock-server to create in-memory http source and sink for DPF HTTP extension and assertions.
- Updated DPF HttpTestFixtures to reuse for UTs and Systems tests.

### CODE REVIEW

File: extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSourceFactory.java [hunk 1]
```diff
@@ -86,17 +86,27 @@ public DataSource createSource(DataFlowRequest request) {
...
+            if (queryParams != null) {
+                httpDataSourceBuilder.queryParams(queryParams);
+            }
+
+            httpDataSourceBuilder
                     .requestId(request.getId())
                     .retryPolicy(retryPolicy)
-                    .monitor(monitor)
-                    .build());
+                    .monitor(monitor);
```
Comment (algattik) [inline extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSourceFactory.java:106]: nit: move up so all non-optional builder calls are done in one train?

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,696 @@
...
+                Arguments.of(HttpStatusCode.FOUND_302.name(), HttpStatusCode.FOUND_302),
+                Arguments.of(HttpStatusCode.BAD_REQUEST_400.name(), HttpStatusCode.BAD_REQUEST_400),
+                Arguments.of(HttpStatusCode.UNAUTHORIZED_401.name(), HttpStatusCode.UNAUTHORIZED_401),
+                Arguments.of(HttpStatusCode.NOT_FOUND_404.name(), HttpStatusCode.NOT_FOUND_404),
+                Arguments.of(HttpStatusCode.INTERNAL_SERVER_ERROR_500.name(), HttpStatusCode.INTERNAL_SERVER_ERROR_500),
+                Arguments.of(HttpStatusCode.BAD_GATEWAY_502.name(), HttpStatusCode.BAD_GATEWAY_502)
+        );
+    }
+
+}
```
Comment (algattik) [inline extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java:696]: is a 700 line class really needed?

File: settings.gradle.kts [hunk 3]
```diff
@@ -171,11 +173,8 @@ include(":samples:04.2-modify-transferprocess:consumer")
 include(":samples:04.2-modify-transferprocess:watchdog")
 include(":samples:04.2-modify-transferprocess:simulator")
 
-include(":samples:04.3-open-telemetry:consumer")
```
Comment (algattik) [inline settings.gradle.kts:174]: why?

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java [hunk 4]
```diff
@@ -0,0 +1,696 @@
...
+
+        // Wait for transfer to be completed.
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                validateTransferResult(processId)
+        );
+
+        // Verify HTTP Source server called exactly once.
+        httpSourceClientAndServer.verify(
+                givenGetRequest(),
+                VerificationTimes.once()
```
Comment (algattik) [inline extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java:390]: static import?

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java [hunk 5]
```diff
@@ -0,0 +1,696 @@
...
+    private HttpRequest givenGetRequest() {
+        return givenGetRequest(Collections.emptyMap());
+    }
+
+    /**
+     * Mock HTTP GET request with query params for source.
+     *
+     * @return see {@link HttpRequest}
+     */
+    private HttpRequest givenGetRequest(Map<String, String> queryParams) {
```
Comment (algattik) [inline extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java:602]:
```suggestion
    private HttpRequest getRequest(Map<String, String> queryParams) {
```
sounds more fluent

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java (hunk 2 above)
Comment (cpeeyush) [reply extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java:696]: Yeah, its too much. Already looking for idea to reduce it.
Suggestions are welcome.

File: settings.gradle.kts (hunk 3 above)
Comment (cpeeyush) [reply settings.gradle.kts]: Fixed. Seems like was there due some merge.

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java (hunk 4 above)
Comment (cpeeyush) [reply extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java:392]: we have other one import static org.mockserver.matchers.Times.once; for request expectation, that's why calling with class name here.

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java (hunk 5 above)
Comment (cpeeyush) [reply extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java]: Done.

File: extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java (hunk 2 above)
Comment (cpeeyush) [reply extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java:582]: Did some refactoring as discussed.

File: extensions/data-plane/data-plane-http/build.gradle.kts [hunk 6]
```diff
@@ -18,13 +18,16 @@ val rsApi: String by project
...
     `java-library`
+    `java-test-fixtures`
 }
 
 dependencies {
     api(project(":extensions:data-plane:data-plane-spi"))
     implementation(project(":common:util"))
     implementation("com.squareup.okhttp3:okhttp:${okHttpVersion}")
     implementation("net.jodah:failsafe:${jodahFailsafeVersion}")
+    testFixturesImplementation("com.github.javafaker:javafaker:1.0.2")
```
Comment (marcgs) [inline extensions/data-plane/data-plane-http/build.gradle.kts:29]: extract version into `gradle.properties`?

Comment (cpeeyush) [reply extensions/data-plane/data-plane-http/build.gradle.kts:29]: Good suggestion, I will do it now.

File: extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSourceFactory.java (hunk 1 above)
Comment (cpeeyush) [reply extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/pipeline/HttpDataSourceFactory.java]: Update it.

File: extensions/data-plane/data-plane-http/build.gradle.kts (hunk 6 above)
Comment (cpeeyush) [reply extensions/data-plane/data-plane-http/build.gradle.kts]: Update it.

**Review by marcgs — APPROVED** [review]

**Comment by marcgs** [comment]
Nice job!

**Review by algattik — APPROVED** [review]

**Pull request merged**